CONTRIBUTING.md

```diff
@@ -169,3 +169,34 @@ Support branches? [support/]
 Version tag prefix? [] eXist-
 Hooks and filters directory? [.git/hooks]
 ```
+
+
+## Commit Naming
+Commits to eXist-db by developers *must* follow a naming convention. This convention expects a prefix to be prepended to each of your commit messages. The prefixes enable us to better understand the commit and also to generate Change Logs. The prefixes in use are:  
+
+* `[feature]`
+    This should be used when a commit adds a new feature.
+* `[bugfix]`
+    This should be used when a commit addresses a bug or issue.
+* `[refactor]`
+    This should be used when a commit is simply refactoring existing code.
+* `[ignore]`
+    This should be used when code is cleaned up by automated means, e.g. reformatting.
+* `[doc]`
+    This should be used for documentation.
+* `[test]`
+    This should be used when a commit solely contains changes to existing tests or adds further tests.
+* `[ci]`
+    This should be used when a commit solely makes changes to CI configuration.
+
+In addition any commit that addresses a GitHub issues, should have an additonal line in its commit message like:
+```
+Closes https://github.com/eXist-db/exist/issues/<github-issue-number>
+```
+
+For example, here is a correctly formatted commit message:
+
+```
+[bugfix] Store the path of Jar files in each EXPath Package's `.exist/classpath.txt` file relative to the package's `content/` sub-folder.
```

Comment:
1. the subject line is too long, it should be no longer than 50 characters
2. the blank line after the subject is missing
3. since it is a fix the body should use the `fixes` keyword instead of `closes`

```
[fix] path to Jars are relative to content subfolder of package

fixes #4901
Store the path of Jar files in each EXPath Package's `.exist/classpath.txt` file relative to the package's `content/` sub-folder.

Reply:
I completely disagree with all 3 points. Please see the existing commit messages from the team over the last several years for existing examples.

Reply:
I'm for the points 1 & 2 as they are also supported by a lot of git utilities and being proposed by [git-scm](https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project).

Reply:
Thanks @reinhapa that link is useful, particularly the [Commit Guidelines](https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines) section. Having read that, I have no objections to @line-o's points 1 and 2 now.